### DISCUSSION TITLE
Vercel Hosted app in NextJS 13 can't handle more than 400 users live...

### DISCUSSION THREAD

**Discussion by FyeVirstate** [vercel/community#3509] [Help]
### Summary

Hi everyone, 

I just created a new website, in 1 week, it's a regular app, using api call to get info from databases, and upload things like this to firebase, and my postgres vercel storage ...

When i have too much users i got this type of error : 
API request error: PrismaClientKnownRequestError: 
Invalid `prisma.historyCloning.findFirst()` invocation:

Timed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 10, connection limit: 5)
    at Rn.handleRequestError (/var/task/node_modules/@prisma/client/runtime/library.js:174:7325)
    at Rn.handleAndLogRequestError (/var/task/node_modules/@prisma/client/runtime/library.js:174:6754)
    at Rn.request (/var/task/node_modules/@prisma/client/runtime/library.js:174:6344)
    at async POST (/var/task/.next/server/app/api/generate/route.js:243:41)
    at async /var/task/.next/server/chunks/5981.js:2702:37 {
  code: 'P2024',
  clientVersion: '4.16.1',
  meta: { connection_limit: 5, timeout: 10 }
}

Also : 
![image](https://github.com/vercel/community/assets/129902255/f18f19a7-2e2c-44b9-98b6-836516500af8)
Multiple 500 error from that ...

I have the Pro Plan, and i just want to know how can i scale to maintain more than 1000 users ?

My code is maybe not so much optimized but i had the same code, on a another hosting server, and it works with 1000 users live... 

There is no real step, it's about my hosting, 

Do you had any experienced like that ? 

I'm waiting to get info to get Entreprise plan...

How to increase our postgres timetout on Vercel storage ?

### Steps to Reproduce

There is no real step, it's about my hosting, 

Do you had any experienced like that ? 

I'm waiting to get info to get Entreprise plan...

How to increase our postgres timetout on Vercel storage ?

**Comment by FyeVirstate** [comment]
Just in case i'm Pro Plan, 

I want to be Entreprise but, the service didn't respond yet...

**Comment by amyegan** [comment]
This connection limit of 100 can be surpassed if you connect to your database with [a connection pooler](https://vercel.com/docs/storage/vercel-postgres/sdk#createclient). Using the pooled connection when deploying your application enables Vercel Postgres to handle up to 10,000 simultaneous connections. You can learn more about it at https://vercel.com/docs/storage/vercel-postgres/faq#pooled-vs.-direct-connections

If you want more info about Enterprise plans, please [contact our Sales team](https://vercel.com/contact/sales). They are always happy to discuss plan options 🙂
